week3_practice/src/components/Board.jsx

```diff
@@ -0,0 +1,48 @@
+import React, { useState, useEffect } from 'react';
+import Tile from './Tile';
+import './Board.css';
+
+const Board = ({ nextNumber, onCorrectClick, onGameComplete }) => {
+  const [numbers, setNumbers] = useState([]);
+
+  useEffect(() => {
+    initializeBoard();
+  }, []);
+
+  const initializeBoard = () => {
+    const initialNumbers = Array.from({ length: 9 }, (_, i) => i + 1); // 1 to 9
+    shuffleArray(initialNumbers);
+    setNumbers(initialNumbers);
+  };
+
+  const shuffleArray = (array) => {
+    for (let i = array.length - 1; i > 0; i--) {
+      const j = Math.floor(Math.random() * (i + 1));
+      [array[i], array[j]] = [array[j], array[i]];
+    }
+  };
```

Comment:
shuffleArray에서 initialNumbers라는 배열의 값을 직접 변경해주면서 불변성이 지켜지지 않고 있습니다..!

React에서는 불변성을 지키는 것이 중요합니다.
물론 이 코드에서는 원본 배열을 다시 사용하지 않기 때문에 문제가 발생하지 않지만, 만약 원본 배열이 다른 곳에서도 사용되고 있다고 가정한다면 직접 값을 바꾸면 다른 곳에도 영향을 주기 때문에 예상치못한 문제가 발생할 수 있습니다.
[참고하면 좋은 블로그 같이 첨부드립니다.](https://devbksheen.tistory.com/entry/%EB%B6%88%EB%B3%80%EC%84%B1%EC%9D%84-%EC%A7%80%ED%82%A4%EB%A9%B4%EC%84%9C-%EA%B0%9D%EC%B2%B4%EC%99%80-%EB%B0%B0%EC%97%B4-%EC%97%85%EB%8D%B0%EC%9D%B4%ED%8A%B8%ED%95%98%EA%B8%B0#%EB%B6%88%EB%B3%80%EC%84%B1%EC%9D%84%20%EC%A7%80%EC%BC%9C%EC%95%BC%20%ED%95%98%EB%8A%94%20%EC%9D%B4%EC%9C%A0-1)

따라서 아래처럼 코드를 수정해서 불변성을 지키는 것은 어떨까요??ㅎㅎ
```suggestion
  const initializeBoard = () => {
    const initialNumbers = Array.from({ length: 9 }, (_, i) => i + 1); // 1 to 9
   const shuffledNumbers = shuffleArray(initialNumbers); 
  setNumbers(shuffledNumbers);
  };

  const shuffleArray = (array) => {
    const newArray = [...array]; // 원본 배열 복사
    for (let i = newArray.length - 1; i > 0; i--) {
      const j = Math.floor(Math.random() * (i + 1));
      [newArray[i], newArray[j]] = [newArray[j], newArray[i]];
    }
    return newArray;
  };
```

Reply:
저도 리액트 첫 과제 했을때 불변성에 대해 코드리뷰 받았었는데, 온전히 이해하기 어려웠던 기억이 새록새록하네요,,!💜🫧
앞으로 계속 신경써야하는 부분이라 참고된 아티클 정독 추천합니다!!